api/src/main/java/org/commonjava/indy/data/DefaultStoreValidator.java

```diff
@@ -0,0 +1,173 @@
+                        );
+                        // .apache.org , 10.192. , .maven.redhat.com
+                        if(allowedNonSSLHostname(remoteHost,host)) {
+                            errors.put(StoreValidationConstants.ALLOWED_SSL,remoteUrl.get().toString());
+                            LOGGER.warn(
+                                "NON-SSL RemoteRepository with URL: "+ host +" is ALLOWED under RULE: " + remoteHost
+                            );
+                        }
+                    }
+                }
+                // Execute HTTP GET & HEAD requests in separate thread pool from executor service
+                Future<Integer> httpGetStatus = executeGetHttp(new HttpGet(remoteUrl.get().toURI()), httpRequestsLatch);
+                Future<Integer> httpHeadStatus = executeHeadHttp(new HttpHead(remoteUrl.get().toURI()), httpRequestsLatch);
+                // Waiting for Http GET & HEAD Request Executor tasks to finish
+                httpRequestsLatch.await();
+                errors.put(StoreValidationConstants.HTTP_GET_STATUS, httpGetStatus.get().toString());
+                errors.put(StoreValidationConstants.HTTP_HEAD_STATUS, httpHeadStatus.get().toString());
+                if(!remoteUrl.get().getProtocol().equalsIgnoreCase(StoreValidationConstants.HTTPS)) {
+                    errors.put(StoreValidationConstants.HTTP_PROTOCOL, remoteUrl.get().getProtocol());
+                }
+                // Check for Sucessfull Validation
+                if (httpGetStatus.get() < 400 && httpHeadStatus.get() < 400) {
+                    LOGGER.warn("=> Success HTTP GET and HEAD Response from Remote Repository: " + remoteUrl.get());
+                    return new ArtifactStoreValidateData
+                        .Builder(remoteRepository.getKey())
+                        .setRepositoryUrl(remoteUrl.get().toExternalForm())
+                        .setValid(true)
+                        .setErrors(errors)
+                        .build();
+                }
+            }
+        }
+        catch (MalformedURLException mue) {
+            LOGGER.error("=> Mailformed URL: ", mue);
+            errors.put(StoreValidationConstants.MAILFORMED_URL,mue.getMessage());
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .setRepositoryUrl( remoteUrl.get().toExternalForm() )
+                .setErrors(errors)
+                .build();
+        }
+        catch (Exception e) {
+            LOGGER.error(" => Not Valid Remote Repository, \n => Exception: " + e);
+            errors.put(StoreValidationConstants.GENERAL, e.getMessage());
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .setRepositoryUrl( remoteUrl.get().toExternalForm() )
+                .setErrors(errors)
+                .build();
+        }
+        if (artifactStore instanceof RemoteRepository) {
```

Comment:
As far as I can tell, we can never return a positive validation response for remote repositories.

Reply:
But this is returning "false" validation object for that RemoteRepository - because like you mention previously we should expect all sort of repositories and maybe some of them are completely with messed up values - thats way i'm returning data object with valid set to "false" and that repo url (if it is having it> ) and errors .